core/src/main/java/org/eclipse/hono/util/CredentialsObject.java

```diff
@@ -361,26 +358,15 @@ public static JsonObject emptySecret(final Instant notBefore, final Instant notA
     }
 
     /**
-     * Creates a salted hash for a password.
-     * <p>
-     * Gets the password's UTF-8 bytes, prepends them with the salt (if not {@code null}
-     * and returns the output of the hash function applied to the byte array.
+     * Uses {@link EncodedPassword#encode()} to return encoded password.
      * 
      * @param hashFunction The hash function to use.
      * @param salt The salt to prepend the password bytes with.
      * @param password The password to hash.
-     * @return The hashed password.
-     * @throws NoSuchAlgorithmException if the given hash function is not supported on
-     *           the JVM.
+     * @return The hashed password as a Base64 formatted String.
      */
-    public static byte[] getHashedPassword(final String hashFunction, final byte[] salt, final String password) throws NoSuchAlgorithmException {
-
-        final MessageDigest digest = MessageDigest.getInstance(hashFunction);
-        if (salt != null) {
-            digest.update(salt);
-        }
-        digest.update(password.getBytes(StandardCharsets.UTF_8));
-        return digest.digest();
+    public static String getHashedPassword(final String hashFunction, final byte[] salt, final String password) {
+        return EncodedPassword.encode(hashFunction, salt, password);
```

Comment:
This method's purpose has been to create a password hash as defined by the [Credentials API](https://www.eclipse.org/hono/api/credentials-api/#hashed-password). Now it seems to create a String of undefined format (neither this method's nor EncodedPassword.encode's JavaDoc even mention the produced format).

I still do not understand what the purpose of this change is, so please either elaborate or revert the change.

Reply:
I don't understand what the problem is. The method used to return byte array that we manually encoded to base64

Base64.getEncoder().encodeToString(getHashedPassword(hashAlgorithm, salt, password)))

Now it returns Base64 encoded String. Nothing else changed.

Reply:
The method originally was used to create a password hash as it would be returned from the Credentials service. This is obviously no longer the case, so I wonder why we no longer need to perform tests based on the hashed password format that the Credentials API defines but instead do tests based on a hash format that has no relevance within Hono.